### PR TITLE
[WIP] Custom CommandElements

### PR DESCRIPTION
**Pull request by boformer**

### CODE REVIEW

**Comment by BlackGoat** [comment]
Remove the WIP in the title when you're done with this. I have no idea what the plan here is.

**Comment by boformer** [comment]
I will finish it this week. It is not a high priority addition.

File: source/plugin/basics/commands/arguments.rst [hunk 1]
```diff
@@ -146,7 +146,74 @@ Overview of the ``GenericArguments`` command elements
...
-.. tip::
+Flags
+=====
+
+Coming soon...
+
+Custom Command Elements
+=======================
+
+It is possible to create custom command elements (e.g. an URL parser or a ``Vector2i`` element).
```
Comment (Mumfrey) [inline source/plugin/basics/commands/arguments.rst]: an -> a

**Comment by RobertHerhold** [comment]
@boformer ETA on flags documentation?

File: source/plugin/basics/commands/arguments.rst [hunk 2]
```diff
@@ -146,7 +146,98 @@ Overview of the ``GenericArguments`` command elements
...
+           String xInput = args.next();
+           int x = parseInt(xInput);
+           
+           String yInput = args.next();
+           int y = parseInt(yInput);
+           
+           return new Vector2i(x, y);
+       }
+       
+       private int parseInt(string input) throws ArgumentParseException {
```
Comment (JBYoshi) [inline source/plugin/basics/commands/arguments.rst]: Capital S in `String`

**Comment by BlackGoat** [comment]
Is there some prospect of this being completed soon, @boformer ?
It is starting to get stale, and already needs a rebase...

**Comment by boformer** [comment]
After fixing the merge conflicts and the small error, it is ready for merge. More examples would be nice, but I can't think of any right now.

**Pull request closed without merging**